lib/initialize_with_config.rb

```diff
@@ -0,0 +1,39 @@
+# frozen_string_literal: true
+
+require_relative "meta"
+require "active_support"
+require "active_support/core_ext"
+
+module InitializeWithConfig
```

Comment:
Can we have rspec tests for this config module as it's a lot more happening here?

Reply:
Rspec tests added for it. I've used the principle of least surprise for overriding methods. If we need to go any farther than that, I'd turn it into a gem.